File: sevent_ lesson_intermediate_certifications/Java_ok/Man.java
import java.time.LocalDate;

public class Man extends Person {
    public boolean hasBeard = true;
    public boolean isHasBeard() {
        return hasBeard;
    }

    public void setHasBeard(boolean hasBeard) {
        this.hasBeard = hasBeard;
    }

    protected String helloPhrase = "Привет, привет!";
    
    public String getHelloPhrase() {
        return helloPhrase;
    }

    
    public void setHelloPhrase(String helloPhrase) {
        this.helloPhrase = helloPhrase;
    }

    public Man(String name) {
        super(name);
    }

    public Man(String name, LocalDate birthday) {
        super(name, birthday);
    }

    @Override
    public void SayHello() {
        System.out.printf("%s: Привет, я мужчина!\n", name);
    }

    public void SayHelloLikeAParent() {
        super.SayHello();
    }

    public void Shave() {
        System.out.printf("%s: Бреется\n", name);
        this.hasBeard = false;
    }

    @Override
    protected void TakeCareImlementation() {
        System.out.printf("%s проверяет уроки и потом идет с детьми на прогулку.\n", name);
    }

    @Override
    public void Print()
    {
        System.out.printf("Имя - %s, день рождения - %s\n", name, birthday);
    }
}

File: sevent_ lesson_intermediate_certifications/C_Sharp_ok/ConsoleApp1/Man.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApp1
{
    internal class Man : Person
    {
        public bool HasBeard { get; private set; } = true;
        protected override string HelloPhrase { get; set; } = "Привет, привет!";

        public Man(string name) : base(name)
        {
        }

        public Man(string name, DateTime birthday) : base(name, birthday)
        {
        }

        public override void SayHello()
        {
            Console.WriteLine($"{Name}: Привет, я мужчина!");
        }

        public void SayHelloLikeAParent()
        {
            base.SayHello();
        }

        public void Shave()
        {
            Console.WriteLine($"{Name}: Бреется");
            this.HasBeard = false;
        }
        protected override void TakeCareImlementation()
        {
                Console.WriteLine($"{Name} проверяет уроки и потом идет с детьми на прогулку.");
        }
        new public void Print()
        {
            Console.WriteLine($"Имя = {Name}, день рождения = {Birthday.ToString("dd.MM.yyyy")}(Метод класса Man)");
        }
    }
}
